File: java/Bubble.java
public class Bubble {
	public static void sort(int a[]) {
		for(int i = 0; i < a.length; i++) {
			boolean swapped = false;
			for(int j = a.length-1; j > i; j--) {
				if(a[j] < a[j-1]) {
					Util.swap(a, j, j-1);
					swapped = true;
				}
			}
			if(!swapped) {
				break;
			}
		}
	}
}

File: php/bubble.php
<?php
require_once('util.php');

// bubble implements bubble sort algorithm. The function takes an array.
function bubble(&$a) {
	for($i = 0; $i < count($a); $i++) {
		$swapped = false;
		for($j = count($a)-1; $j > $i; $j--) {
			if($a[$j] < $a[$j-1]) {
				swap($a, $j, $j-1);
				$swapped = true;
			}
		}
		if(!$swapped) {
			break;
		}
	}
}
